### PR TITLE
use native_layer_norm_backward decomp from pytorch core

### PR DESCRIPTION
**Pull request by bdhirsh**
We want to make `native_layer_norm` `CompositeExplicitAutograd` in core (explanation in the core PR: https://github.com/pytorch/pytorch/pull/78183), but doing so requires xla to have a kernel it can use for `native_layer_norm_backward).

We actually have one, but we need to access it through torchscript instead of the normal dispatcher machinery. I added the necessary bits of boilerplate + a note on why here.

### CODE REVIEW

**Comment by byronyi** [comment]
Does this have performance implications? Before this PR, layer norm is implemented using batch norm, and autograds to batch norm backward in XLA. After this PR we have a bunch of [smaller operators](https://github.com/pytorch/pytorch/blob/master/torch/_decomp/decompositions.py#L921-L994) that does not necessarily get fused in XLA compiler.

**Comment by bdhirsh** [comment]
@byronyi just to be safe, I ended up updating this PR to do the lower-delta change and just re-use the existing forward decomps from core (which will directly call `native_batch_norm`/`native_batch_norm_backward`).

It would be nice at some point to benchmark and see if XLA can actually recover the perf from the full decomposition though :)

**Comment by bdhirsh** [comment]
cc @JackCaoG. the tldr is that we technically do have "math composite" kernels for layer/group norm in core, but at least in core, we don't want to treat them as "autograd composite", because we want function transforms (functionalization + vmap) to treat them as primitives.

It seems ok for backends to choose to re-use the decomposition for autograd though, which is what this PR does for XLA.

**Comment by JackCaoG** [comment]
@bdhirsh Does that means in order to use functionzation pass in ltc in the future, we need to make some follow up change as well?

**Review by JackCaoG — APPROVED** [review]
Thanks @bdhirsh !

**Comment by JackCaoG** [comment]
Thanks @bdhirsh !
